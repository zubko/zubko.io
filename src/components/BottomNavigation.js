/**
 * Component which displays Next/Prev navigation at the bottom of the page
 *
 * @flow
 */
import React from 'react';
import styled from '@emotion/styled';

import { Link } from '../components';
import { rhythm } from '../Typography';

type Item = {
  path: string,
  title: string,
};

type Props = {
  next?: ?Item,
  prev?: ?Item,
  back?: ?Item,
};

export default ({ next, prev, back }: Props) => (
  <>
    <hr />
    <OptionContainer>
      {prev && (
        <span>
          <OptionTitle>Prev:</OptionTitle> <br />
          <Link to={prev.path}>← {prev.title}</Link>
        </span>
      )}
    </OptionContainer>
    <OptionContainer>
      {next && (
        <span>
          <OptionTitle>Next:</OptionTitle> <br />
          <Link to={next.path}>{next.title} →</Link>
        </span>
      )}
    </OptionContainer>
    <OptionContainer>
      {back && (
        <span>
          <OptionTitle>Back:</OptionTitle> <br />
          <Link to={back.path}>← {back.title}</Link>
        </span>
      )}
    </OptionContainer>
  </>
);

const OptionContainer = styled.div`
  margin-bottom: ${rhythm(1)};
`;

const OptionTitle = styled.span`
  font-weight: 600;
`;
