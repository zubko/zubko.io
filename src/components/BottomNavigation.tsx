import styled from "@emotion/styled";

import { rhythm } from "../Typography";
import Link from "./Link";

type Item = {
  path: string;
  title: string;
};

type Props = {
  next?: Item;
  prev?: Item;
  back?: Item;
};

/**
 * Component which displays Next/Prev navigation at the bottom of the page
 */
export default ({ next, prev, back }: Props) => (
  <>
    <hr />
    <OptionContainer>
      {prev && (
        <span>
          <OptionTitle>Prev:</OptionTitle> <br />
          <Link to={prev.path}>← {prev.title}</Link>
        </span>
      )}
    </OptionContainer>
    <OptionContainer>
      {next && (
        <span>
          <OptionTitle>Next:</OptionTitle> <br />
          <Link to={next.path}>{next.title} →</Link>
        </span>
      )}
    </OptionContainer>
    <OptionContainer>
      {back && (
        <span>
          <OptionTitle>Back:</OptionTitle> <br />
          <Link to={back.path}>← {back.title}</Link>
        </span>
      )}
    </OptionContainer>
  </>
);

const OptionContainer = styled.div`
  margin-bottom: ${rhythm(1)};
`;

const OptionTitle = styled.span`
  font-weight: 600;
`;
