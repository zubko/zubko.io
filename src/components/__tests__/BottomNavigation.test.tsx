import React from 'react';
import renderer from 'react-test-renderer';

import BottomNavigation from '../BottomNavigation';

describe('BottomNavigation', () => {
  it('renders without regressions', () => {
    const tree = renderer
      .create(
        <BottomNavigation
          prev={{ title: 'prev-title', path: '/prev-path' }}
          next={{ title: 'next-title', path: '/next-path' }}
          back={{ title: 'back-title', path: '/back-path' }}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
