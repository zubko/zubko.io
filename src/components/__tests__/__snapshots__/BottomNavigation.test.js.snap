// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BottomNavigation renders without regressions 1`] = `
Array [
  <hr />,
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": ".css-ldhagr{margin-bottom:1.618rem;}",
      }
    }
    data-emotion="css ldhagr"
  />,
  <div
    className="css-ldhagr"
  >
    <span>
      <style
        dangerouslySetInnerHTML={
          Object {
            "__html": ".css-66vicj{font-weight:600;}",
          }
        }
        data-emotion="css 66vicj"
      />
      <span
        className="css-66vicj"
      >
        Next:
      </span>
       
      <br />
      <a
        css={
          Object {
            "map": undefined,
            "name": "1k30ml3",
            "next": undefined,
            "styles": "
  color: #2C8127;
  text-decoration: none;

  background-image: linear-gradient(
    to bottom,
    currentColor 0%,
    currentColor 100%
  );
  background-position: 0 100%;
  background-repeat: repeat-x;
  background-size: 4px 1px;
  transition: background-size 0s;

  :hover {
    color: black;
    text-decoration: none;
    background-image: linear-gradient(
      to bottom,
      #bcebb9 0%,
      #bcebb9 100%
    );
    background-size: 4px 100%;
    transition: background-size 0.15s;
  }
",
            "toString": [Function],
          }
        }
        href="/next-path"
      >
        next-title
         →
      </a>
    </span>
  </div>,
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": ".css-ldhagr{margin-bottom:1.618rem;}",
      }
    }
    data-emotion="css ldhagr"
  />,
  <div
    className="css-ldhagr"
  >
    <span>
      <style
        dangerouslySetInnerHTML={
          Object {
            "__html": ".css-66vicj{font-weight:600;}",
          }
        }
        data-emotion="css 66vicj"
      />
      <span
        className="css-66vicj"
      >
        Prev:
      </span>
       
      <br />
      <a
        css={
          Object {
            "map": undefined,
            "name": "1k30ml3",
            "next": undefined,
            "styles": "
  color: #2C8127;
  text-decoration: none;

  background-image: linear-gradient(
    to bottom,
    currentColor 0%,
    currentColor 100%
  );
  background-position: 0 100%;
  background-repeat: repeat-x;
  background-size: 4px 1px;
  transition: background-size 0s;

  :hover {
    color: black;
    text-decoration: none;
    background-image: linear-gradient(
      to bottom,
      #bcebb9 0%,
      #bcebb9 100%
    );
    background-size: 4px 100%;
    transition: background-size 0.15s;
  }
",
            "toString": [Function],
          }
        }
        href="/prev-path"
      >
        ← 
        prev-title
      </a>
    </span>
  </div>,
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": ".css-ldhagr{margin-bottom:1.618rem;}",
      }
    }
    data-emotion="css ldhagr"
  />,
  <div
    className="css-ldhagr"
  >
    <span>
      <style
        dangerouslySetInnerHTML={
          Object {
            "__html": ".css-66vicj{font-weight:600;}",
          }
        }
        data-emotion="css 66vicj"
      />
      <span
        className="css-66vicj"
      >
        Back:
      </span>
       
      <br />
      <a
        css={
          Object {
            "map": undefined,
            "name": "1k30ml3",
            "next": undefined,
            "styles": "
  color: #2C8127;
  text-decoration: none;

  background-image: linear-gradient(
    to bottom,
    currentColor 0%,
    currentColor 100%
  );
  background-position: 0 100%;
  background-repeat: repeat-x;
  background-size: 4px 1px;
  transition: background-size 0s;

  :hover {
    color: black;
    text-decoration: none;
    background-image: linear-gradient(
      to bottom,
      #bcebb9 0%,
      #bcebb9 100%
    );
    background-size: 4px 100%;
    transition: background-size 0.15s;
  }
",
            "toString": [Function],
          }
        }
        href="/back-path"
      >
        ← 
        back-title
      </a>
    </span>
  </div>,
]
`;
